crates/fluvio/src/stats/data_point.rs

```diff
@@ -0,0 +1,316 @@
+    }
+
+    /// Return the throughput of the session
+    pub fn total_throughput(&self) -> DataThroughput {
+        Self::covert_to_largest_throughput_unit(self.total_bytes() / self.uptime())
+    }
+
+    /// Return the pid of the client
+    pub fn pid(&self) -> u32 {
+        self.pid
+    }
+
+    /// Returns the offset last seen
+    pub fn offset(&self) -> i32 {
+        self.offset
+    }
+
+    /// Returns the last data transfer size in bytes
+    pub fn last_bytes(&self) -> DataVolume {
+        self.last_bytes
+    }
+
+    /// Returns the accumulated data transfer size in bytes
+    pub fn total_bytes(&self) -> DataVolume {
+        self.total_bytes
+    }
+
+    /// Returns the latency of last transfer -> Duration {
+    pub fn last_latency(&self) -> QuantDuration {
+        self.last_latency
+    }
+
+    /// Returns the last memory usage sample
+    pub fn memory(&self) -> DataVolume {
+        self.last_mem
+    }
+
+    /// Returns the last cpu usage sample as a percentage (%)
+    pub fn cpu(&self) -> f32 {
+        self.last_cpu
+    }
+
+    /// Returns the current uptime of the client
+    pub fn uptime(&self) -> QuantDuration {
+        let time_elapsed = unix_timestamp_nanos() - self.start_time;
+        let scalar: AmountT = (time_elapsed as f32).into();
+        Self::convert_to_largest_time_unit(scalar * NANOSECOND)
+    }
+
+    /// Returns the number of records transferred
+    pub fn records(&self) -> u64 {
+        self.num_records
+    }
+
+    /// Generates a CSV header string. Columns returned based on client's configured `ClientStatsDataCollect`
+    pub fn csv_header(&self) -> Result<String, FluvioError> {
+        if self.stats_collect() == ClientStatsDataCollect::None {
+            return Err(FluvioError::Other(
+                "Client is not collecting stats".to_string(),
+            ));
+        }
+
+        let mut header = vec![
+            "pid".to_string(),
+            "start_time_ns".to_string(),
+            "last_updated_ns".to_string(),
+        ];
+
+        let data_related = vec![
+            "offset".to_string(),
+            "total_records".to_string(),
+            "total_bytes".to_string(),
+            "last_latency_ns".to_string(),
+            "last_bytes".to_string(),
+            "throughput_kbps".to_string(),
+        ];
+
+        let system_related = vec!["last_mem_kb".to_string(), "last_cpu_pct".to_string()];
+
+        if (self.stats_collect() == ClientStatsDataCollect::All)
+            || (self.stats_collect() == ClientStatsDataCollect::Data)
+        {
+            header.extend(data_related);
+        }
+
+        if (self.stats_collect() == ClientStatsDataCollect::All)
+            || (self.stats_collect() == ClientStatsDataCollect::System)
+        {
+            header.extend(system_related);
+        }
+
+        // Render header as CSV w/ trailing newline
+        header.push("\n".to_string());
```

Comment:
These two lines end up creating the header line with a trailing comma.  Appending the "\n" to the joined string would avoid that.